### PR TITLE
Document `Flop Counter Mode` in torch.utils

### LINKED ISSUE #145555: need to document `FlopCounterMode`
### 📚 The doc issue

`FlopCounterMode` as demo'ed here https://gist.github.com/Chillee/07b36672a0ca2d1280e42b8d10f23174 needs to be documented please.

And thank you!

cc: @Chillee 

### Suggest a potential alternative/fix

_No response_

### PR DESCRIPTION
**Pull request by ParagEkbote**
Fixes #145555 

As discussed in the issue, I've added a brief introduction to `torch.utils` and only added the functions which have been documented in the Flop Counter module. Could you please review?

@pytorchbot label "module: docs" "topic: docs" "topic: not user facing"

cc @svekars @sekyondaMeta @AlannaBurke

### CODE REVIEW

File: docs/source/utils.md [hunk 1]
```diff
@@ -3,6 +3,23 @@
...
+    :toctree: generated
+    :nosignatures:
+
+    conv_flop_count
+    conv_backward_flop
+    sdpa_flop_count
+    _offsets_to_lengths
+    _unpack_flash_attention_nested_shapes
+    _unpack_efficient_attention_nested_shapes
+
```
Comment (fffrog) [inline docs/source/utils.md]: Move those changes into Line 26 please

Comment (ParagEkbote) [reply docs/source/utils.md]: I have done the changes. Could you please review?

**Comment by ParagEkbote** [comment]
Closing this PR for now.
